app/src/main/java/ch/epfl/sdp/GameInfoActivity.java

```diff
@@ -0,0 +1,76 @@
+package ch.epfl.sdp;
+
+import android.os.Bundle;
+import android.widget.TextView;
+import java.util.*;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import static java.lang.Math.toRadians;
+
+public class GameInfoActivity extends AppCompatActivity {
+
+    Player player = new Player(6.149290,
+            46.212470,
+            50,
+            "admin",
+            "admin@epfl.ch");
+
+    TextView username, healthPoint;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_game_info);
+
+        username = findViewById(R.id.username_text);
+        username.setText(String.valueOf(player.getUsername()));
+
+        healthPoint = findViewById(R.id.healthpoint_text);
+        healthPoint.setText(String.valueOf(player.getHealthPoints()));
+
+
+        // Update value of health point every 0.1 second
+        Thread thread = new Thread() {
```

Comment:
Maybe this should be called on an event ( for now to illustrate, maybe on a button again ). It would solve the code climate issue about the oncreate method. It would also fit more into a class than an activity I think.

Reply:
A good suggestion, I agree with you that using an event may save more resources than refresh frequently. But I think we can leave it to refresh every 0.1 second at this PR and later create another one to make it to be an event in order to save the resources. As to solve the code climate issue, I refactor the code, hope we can pass now.